pkg/git/k8s_git_keychain.go

```diff
@@ -28,15 +29,40 @@ func newK8sGitKeychain(k8sClient k8sclient.Interface) *k8sGitKeychain {
 }
 
 func (k *k8sGitKeychain) Resolve(namespace, serviceAccount string, git v1alpha1.Git) (transport.AuthMethod, error) {
-	creds, err := k.secretManager.SecretForServiceAccountAndURL(serviceAccount, namespace, git.URL)
-	if err != nil && !k8serrors.IsNotFound(err) {
+	endpoint, err := transport.NewEndpoint(git.URL)
```

Comment:
I really like this  'transport.Endpoint' object. I feel like it would make a good foundation for a true git keychain interface.